h2o-extensions/xgboost/src/main/java/biz/k11i/xgboost/tree/NodeHelper.java

```diff
@@ -0,0 +1,27 @@
+package biz.k11i.xgboost.tree;
+
+import biz.k11i.xgboost.util.ModelReader;
+
+import java.io.IOException;
+
+/**
+ * This class exposes some package-private APIs of RegTreeImpl.Node and provides additional helper methods.
+ * These methods can eventually be folded back into the original class.
+ */
+public class NodeHelper {
+
+    public static RegTreeNode read(ModelReader reader) throws IOException {
+        return new RegTreeImpl.Node(reader);
+    }
+
+    public static boolean isEqual(RegTreeNode left, RegTreeNode right) {
+        return left.getParentIndex() == right.getParentIndex() 
```

Comment:
Shouldn't this method also expect that `left` or `right` could be `null`?

Reply:
yes, in this context null, can be considered - I will add that

Reply:
fixed